src/mesh/Mesh.cpp

```diff
@@ -466,5 +466,139 @@ std::ostream &operator<<(std::ostream &os, const Mesh &m)
   return os;
 }
 
+bool Mesh::computeQuadConvexityFromPoints(std::array<int,4> &vertexIDs) const
+{
+  /*
+    All points need to be projected into a new plane with only 2 coordinates, x' and y'. These are used to check 
+    the convexity of the quad. These new coordinates are stored in 'coords'. Vertices in 2D does not need to
+    be transformed.
+  */
+  Eigen::Vector3d coords[4];
+
+  // Normal of the plane of first three points in the list of vertices
+  Eigen::Vector3d e_1 = vertices()[vertexIDs[1]].getCoords() - vertices()[vertexIDs[0]].getCoords();
+  Eigen::Vector3d e_2 = vertices()[vertexIDs[2]].getCoords() - vertices()[vertexIDs[0]].getCoords();
+  Eigen::Vector3d normalVector = e_1.cross(e_2);
+
+  //Transform Coordinates - coord[0] is the origin
+  for (int i = 0; i < 4; i++){
+    Eigen::Vector3d distance = vertices()[vertexIDs[i]].getCoords() - vertices()[vertexIDs[0]].getCoords();
+    coords[i][0] = e_1.dot(distance);
+    coords[i][1] = e_2.dot(distance);
+    coords[i][2] = normalVector.dot(distance);
```

Comment:
```suggestion
    Eigen::Vector3d difference = vertices()[vertexIDs[i]].getCoords() - vertices()[vertexIDs[0]].getCoords();
    coords[i][0] = e_1.dot(difference);
    coords[i][1] = e_2.dot(difference);
    coords[i][2] = normalVector.dot(difference);
```
I am also not really happy with `difference`, but is sounds more like a vector than `distance`. Could not find a better name right now.

Reply:
euclideanDistance would be the most accurate?